pallets/data-preservers/src/lib.rs

```diff
@@ -151,6 +271,246 @@ pub mod pallet {
 
             Ok(())
         }
+
+        #[pallet::call_index(1)]
+        #[pallet::weight(T::WeightInfo::create_profile(
+            profile.url.len() as u32,
+            profile.limited_to_para_ids.as_ref().map(|v| v.len() as u32).unwrap_or(0)
+        ))]
+        pub fn create_profile(
+            origin: OriginFor<T>,
+            profile: Profile<T>,
+        ) -> DispatchResultWithPostInfo {
+            let account = ensure_signed(origin)?;
+
+            let deposit = T::ProfileDeposit::profile_deposit(&profile)?;
+            T::Currency::hold(&HoldReason::ProfileDeposit.into(), &account, deposit)?;
+
+            let id = NextProfileId::<T>::get();
+            NextProfileId::<T>::set(id.checked_add(1).ok_or(ArithmeticError::Overflow)?);
+
+            ensure!(
+                !Profiles::<T>::contains_key(id),
+                Error::<T>::NextProfileIdShouldBeAvailable
```

Comment:
This would be a very bad thing right? It would mean that something goes really wrong. In that case can we add a extrinsic to set the next profile id?

Reply:
If it were to actually overflow it would be very bad, thus why I protect for it. However it shouldn't overflow with the proper Id type. As you suggested above I can make ProfileId generic and configurable, however one must be carefull it will not realistically overflow.

Reply:
`ParaId` is a u32 in the relay chain and it has exactly the same implementation, so I wouldn't worry about overflows.

Reply:
let's imagine we code something wrong (e.g., the way we assign profile ids, that at some point is not incremented for whatever reason). In this case no-one else would be able to set a profile id anymore. Hence my suggestion for adding a `force_next_profile_id` that can set it to whatever you want

Reply:
I don't think that's useful, if there's a bug we will release a new runtime with a migration and never use that extrinsic again, and we can just sudo.setStorage instead.

Reply:
fair enough